src/serde.rs

```diff
@@ -120,7 +124,9 @@ pub mod str {
     where
         S: serde::Serializer,
     {
-        value.to_string().serialize(serializer)
+        ArrayString::<MAX_STR_BUFFER_SIZE>::try_from(format_args!("{}", value))
+            .map_err(S::Error::custom)?
+            .serialize(serializer)
```

Comment:
There is no need to perform a heap allocation.

`to_string` comes from `Display`, which is the same of `format_args!("{}", value)`.